### PR TITLE
Fix:updating an user with empty username and name gives an error message

### LINKED ISSUE #598: Updating an user with empty username and name should give an error message
When updating an user with empty username and name should give an error message.

**To Reproduce**

1. Go to 'User-update user profile'.
2. Provide authorization access token.
3. {
  "name": "",
  "username": "",
  "bio": "string",
  "location": "string",
  "occupation": "string",
  "organization": "string",
  "slack_username": "string",
  "social_media_links": "string",
  "skills": "string",
  "interests": "string",
  "resume_url": "string",
  "photo_url": "string",
  "need_mentoring": true,
  "available_to_mentor": true
   }

4. Click on 'Execute'.
5. Scroll down to 'Responses'.
6.  
Screenshot (75)
7. current user's detail after update in" Return details of current user".
Screenshot (76)

**Expected behavior**
If the username and name are not expected to be updated to empty string and error should be shown.

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser chrome

### PR DESCRIPTION
**Pull request by rashikaqureshi**
while updating user details if the username or name is empty. then added an error message. 
Fixes #598
Tested these scenarios. Earlier for these scenarios error message was - user was updated successfully.
 Now populating a different error message. Hence receiving errors.

Screenshot 2020-07-21 21 17 50
Screenshot 2020-07-21 21 20 39
Screenshot 2020-07-21 21 24 26
Screenshot 2020-07-21 21 25 08

**Code/Quality Assurance Only**
- Bugfix (non-breaking change which fixes an issue)
-  My PR follows the style guidelines of this project
-  I have performed a self-review of my own code or materials

**Code/Quality Assurance Only**

- [ ] My changes generate no new warnings 
- [ X] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by vj-codes** [comment]
@anitab-org/coding-team This PR needs review. Thanks!

File: app/api/dao/user.py [hunk 1]
```diff
@@ -223,8 +224,17 @@ def update_user_profile(user_id: int, data: Dict[str, str]):
 
             user.username = username
 
-        if "name" in data and data["name"]:
-            user.name = data["name"]
+        if "name" in data:
+            if data["name"]:
+                user.name = data["name"]
+            else:
+                return messages.NAME_FIELD_IS_MISSING,HTTPStatus.OK
```
Comment (SanketDG) [inline app/api/dao/user.py:231]: I'd rewrite this as:

```python
try:
    user.name = data["name"]
except KeyError:
    return ....
```

**Review by SanketDG — CHANGES_REQUESTED** [review]
Some basic style changes

**Comment by devkapilbansal** [comment]
@rashikaqureshi please do the changes as requested

**Comment by vj-codes** [comment]
@rashikaqureshi Closing the PR due to no update response and inactivity.
Thank you for your contribution 🎉

**Pull request closed without merging**